src/charm.py

```diff
@@ -1146,20 +1169,274 @@ def _addon_reconciliation(self, addon_type):
+        if not result.success:
+            raise exceptions.WordPressBlockedStatusException(
+                f"Unable to config akismet plugin, {result.message}"
+            )
+
+    @staticmethod
+    def _encode_openid_team_map(team_map):
+        """Convert wp_plugin_openid_team_map setting to WordPress openid_teams_trust_list option
+
+        example input: site-sysadmins=administrator,site-editors=editor,site-executives=editor
+        """
+        team_map_lines = []
+        i = 0
+        team_map_lines.append("a:{}:{{".format(len(team_map.split(","))))
+        for mapping in team_map.split(","):
+            i = i + 1
+            team, role = mapping.split("=", 2)
+            team_map_lines.append("i:{};".format(i))
+            team_map_lines.append('O:8:"stdClass":4:{')
+            team_map_lines.append('s:2:"id";')
+            team_map_lines.append("i:{};".format(i))
+            team_map_lines.append('s:4:"team";')
+            team_map_lines.append('s:{}:"{}";'.format(len(team), team))
+            team_map_lines.append('s:4:"role";')
+            team_map_lines.append('s:{}:"{}";'.format(len(role), role))
+            team_map_lines.append('s:6:"server";')
+            team_map_lines.append('s:1:"0";')
+            team_map_lines.append("}")
+        team_map_lines.append("}")
+
+        return "".join(team_map_lines)
+
+    def _plugin_openid_reconciliation(self):
+        """Reconciliation process for the openid plugin"""
+        openid_team_map = self.model.config["wp_plugin_openid_team_map"].strip()
+        if not openid_team_map:
+            result = self._deactivate_plugin(
+                "openid",
+                ["openid_required_for_registration", "openid_teams_trust_list"]
+            )
+        else:
+            result = self._activate_plugin("openid", {
+                "openid_required_for_registration": "1",
+                "openid_teams_trust_list": self._encode_openid_team_map(openid_team_map)
+            })
+        if not result.success:
+            raise exceptions.WordPressBlockedStatusException(
+                f"Unable to config openid plugin, {result.message}"
+            )
+
+    def _apache_config_is_enabled(self, conf_name):
+        """Check if a specified apache configuration file is enabled"""
+        enabled_config = [
+            name for name in self._container().list_files("/etc/apache2/conf-enabled")
+        ]
+        return f"{conf_name}.conf" in enabled_config
+    def _apache_enable_config(self, conf_name, conf):
+        """enables a specified configuration file within the apache2 configuration"""
+        self._stop_server()
+        self._container().push(
+            path=f"/etc/apache2/conf-available/{conf_name}.conf", source=conf
+        )
+        self._run_cli(["a2enconf", conf_name])
+        self._start_server()
+
+    def _apache_disable_config(self, conf_name):
+        """disables a specified configuration file within the apache2 configuration"""
```

Comment:
Can we capitalise the first word in the docstring?

Reply:
Fixed, thanks!